spec-main/api-browser-window-spec.ts

```diff
@@ -638,6 +638,38 @@ describe('BrowserWindow module', () => {
       })
     })
 
+    describe('BrowserWindow.moveAbove(mediaSourceId)', () => {
+      it('should throw an exception if wrong formatting', async () => {
+        const fakeSourceIds = ['none', 'screen:0', 'window:fake', 'window:1234']
+        fakeSourceIds.forEach((sourceId) => {
+          expect(() => {
+            w.moveAbove(sourceId)
+          }).to.throw(/Invalid media source id/)
+        })
+      })
+      it('should throw an exception if invalid window', async () => {
+        // It is very unlikely that these window id exist.
+        const fakeSourceIds = ['window:99999999:0', 'window:123456:1',
+          'window:123456:9']
+        fakeSourceIds.forEach((sourceId) => {
+          expect(() => {
+            w.moveAbove(sourceId)
+          }).to.throw(/Invalid media source id/)
+        })
+      })
+      it('should not throw an exception', async () => {
+        const w2 = new BrowserWindow({ show: false, title: 'window2' })
+        const w2Shown = emittedOnce(w2, 'show')
+        w2.show()
+        await w2Shown
+
+        expect(() => {
+          w.moveAbove(w2.getMediaSourceId())
+        }).to.not.throw()
+
+        await closeWindow(w2, { assertNotWindows: false })
```

Comment:
Is there some way you can test that the window was actually moved?

Reply:
Not really, or maybe with DesktopCapturer.getSources() as they are returned from top to bottom but I can see the flakiness coming.

What we need is an option to `BrowserWindow.getAllWindows()` to get all windows from top to bottom in term of z-order.

Reply:
if `DesktopCapturer.getSources()` is guaranteed to return in the correct z-order then that seems like a reasonable way to test.

Reply:
Oki will do.

Reply:
Hi @nornagon , I added a such test. You will that if the requirements are not met the test is skipped.